src/App.jsx

```diff
@@ -14,33 +15,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const sortGoods = (goods, sortType, isReverse) => {
+  let sortedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  switch (sortType) {
+    case 'alphabet':
+      sortedGoods = [...goods].sort((a, b) => a.localeCompare(b));
+      break;
+    case 'length':
+      sortedGoods = [...goods].sort((a, b) => a.length - b.length);
+      break;
+    default:
+      sortedGoods = [...goods];
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (isReverse) {
+    sortedGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [activeSort, setActiveSort] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const goods = sortGoods(goodsFromServer, activeSort, isReverse);
+
+  const sortAlphabetically = () => {
+    setActiveSort('alphabet');
+  };
+
+  const sortByLength = () => {
+    setActiveSort('length');
+  };
+
+  const goodsReverse = () => {
+    setIsReverse(prev => !prev);
+  };
+
+  const goodsReset = () => {
+    setActiveSort('');
+    setIsReverse(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${activeSort === 'alphabet' ? '' : 'is-light'}`}
+          onClick={sortAlphabetically}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${activeSort === 'length' ? '' : 'is-light'}`}
+          onClick={sortByLength}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${isReverse === true ? '' : 'is-light'}`}
+          onClick={goodsReverse}
+        >
+          Reverse
+        </button>
+
+        {(activeSort !== '' || isReverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={goodsReset}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul>
+        {goods.map(good => (
+          <li data-cy="Good">{good}</li>
```

Comment:
Each list item should have a unique `key` prop to help React identify which items have changed, are added, or are removed. Consider using the index of the map function or a unique property of the `good` as the `key`.